lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/DriverStartHandler.java

```diff
@@ -73,10 +105,18 @@ public void onNext(final StartTime startTime) {
   }
 
   private void onRestart(final StartTime startTime) {
-    if (restartHandlers.isPresent()) {
-      for (EventHandler<StartTime> restartHandler : this.restartHandlers.get()) {
+    if (this.driverRestartManager.isPresent() && this.restartHandlers.isPresent()) {
+      for (EventHandler<StartTime> serviceRestartHandler : this.serviceRestartHandlers.get()) {
+        serviceRestartHandler.onNext(startTime);
+      }
+
+      for (EventHandler<StartTime> restartHandler : this.restartHandlers.get()){
         restartHandler.onNext(startTime);
       }
+
+      // This can only be called after calling client restart handlers because REEF.NET
+      // JobDriver requires making this call to set up the InterOp handlers.
+      this.driverRestartManager.get().onRestartRecoverEvaluators();
```

Comment:
Can you please file a JIRA for that issue and link it here as a `//TODO[REEF-xxx]: ...`?